### ISSUE TITLE
What are the _best network weights?

### ISSUE THREAD

**Issue by rgov** [AlexeyAB/darknet#6762]
The README says to select the weights that perform best on the validation set, to prevent overfitting.

Is this what the `_best` weights file already is? 

Unlike `_last` or `_final`, which are *the same* as one of the numbered weights file, `_best` is a different set of weights than any of the others, so it's hard to tell what epoch/batch number Darknet chose to be the best one. Should I trust it?

**Comment by stephanecharette** [comment]
I had always assumed that ..._best.weights is the one we should be using.  When you sort the weights by time+date, it gives you an approximate idea of when the _best.weights was saved.

But now I'm really curious to see what other people have to say on the subject.

**Comment by ark-** [comment]
My understanding...

When `--map` is enabled, the `*_best.weights` file are the weights which were run on the validation set which gave the highest mAP. 

This calculation is run sporadically during training and can be seen in the log as "Next mAP calculation at iteration: xxx". I think storing the logs would give you an idea of which iteration the highest mAP occurred and then you could test some other weights files around this on your validation sets to determine your own best weights.

**Comment by johnny-mueller** [comment]
I am of the same opinion as @ark-  . It selects the weights with the highest mAP. If you really want to get the best model, you have to calculate the mAP of each stored weight, because Darknet (without changes) calculates the mAP only every 4th epoch. However, in most cases the actual best model is located in the "area around" the model chosen by darknet. At least that's the experience I have gained so far.
